### DISCUSSION TITLE
Make different background colour for lines in logs view for different unique sets of labels.

### DISCUSSION THREAD

**Discussion by avkonst** [grafana/grafana#56845] [Observability]
**What would you like to be added** and **Why is this needed**:: 

Currently all log lines have got the same background regardless of the labels. For example:

![image](https://user-images.githubusercontent.com/3971413/87096739-97bfc880-c297-11ea-96a6-be84f13b89cb.png)

As you can see unique labels are m1, m2, m3, which indicates they are coming from different sources (containers or hosts). These logs are hard to read. It becomes painful for eyes due to very precise focus, especially if you need to deal with these logs all day.

I know it is possible to open separate grafana tabs (one per unique source of logs), but it is even worse as it requires sorting / relating by time across multiple windows,, which kills the purpose of aggregated logs view.

Instead, I propose to add a toggle button, which would enable different backgrounds for log lines. Each set of unique labels would get assigned it's own unique background colour. Ideally, once a colour is assigned to a unique set of labels it is preserved while a browser tab is open. Further step could be making background colours configurable and persistent in settings. 

Does it make sense?

**Comment by avkonst** [comment]
I noticed you applied the label "area/explore". I am guessing it is because it relates to the Explore tab. But it is not only for Explore tab. The same is applicable for configurable logging panels / widgets. It is also applicable to live view mode of the logging panel / Explore tab

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
